app/main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (
+    NotWearingMaskError,
+    VaccineError
+)
+
+
+def go_to_cafe(friends: list[dict], cafe: "Cafe") -> str | Exception:
+    masks_to_buy = 0
+    try:
+        for friend in friends:
+            try:
+                cafe.visit_cafe(friend)
+            except NotWearingMaskError:
+                masks_to_buy += 1
+    except VaccineError:
+        return "All friends should be vaccinated"
```

Comment:
The outer `try-except` block catches `VaccineError`, which means if any friend raises this error, the loop will stop, and the function will return immediately. If the intention is to check each friend independently, consider moving the `VaccineError` handling inside the loop, similar to how `NotWearingMaskError` is handled.